### ISSUE TITLE
Add hidden abilities in the ability pool for gen5 and after

### ISSUE THREAD

**Issue by AdemAFN** [champred/UPR-Android#14]
Hi, it's my first time using github so I'm not sure if I'm doing this right, I apologize in advance if I made a mistake. All about my request is pretty much in the title.

**Comment by champred** [comment]
I'm unsure what you're asking. Could you elaborate? The randomizer already handles hidden abilities here:

https://github.com/champred/UPR-Android/blob/2b638b579eecda4fae63fcb788e4992b6461eee6/upr/src/com/dabomstew/pkrandom/romhandlers/AbstractRomHandler.java#L659-L662

What would the expected behavior of this feature be?

**Comment by AdemAFN** [comment]
> I'm unsure what you're asking. Could you elaborate? The randomizer already handles hidden abilities here:
> 
> https://github.com/champred/UPR-Android/blob/2b638b579eecda4fae63fcb788e4992b6461eee6/upr/src/com/dabomstew/pkrandom/romhandlers/AbstractRomHandler.java#L659-L662
> 
> What would the expected behavior of this feature be?

It would give pokemon the possibility to have their hidden abilities without the randomize abilities setting, like making it possible to naturally get a snivy with contrary

**Comment by champred** [comment]
I don't believe it would be possible for the randomizer to change any of the game mechanics. The hidden ability can only be obtainable from hidden grottos as far as I'm aware. Changing that would require a ROM hack or code modification. The only "alternative" would be to swap either the primary or secondary ability with the hidden ability.

**Comment by AdemAFN** [comment]
> I don't believe it would be possible for the randomizer to change any of the game mechanics. The hidden ability can only be obtainable from hidden grottos as far as I'm aware. Changing that would require a ROM hack or code modification. The only "alternative" would be to swap either the primary or secondary ability with the hidden ability.

There's a setting called "Ensure two abilities" would it work on pokemon who have only one ability like starters ? If yes maybe with something like that

**Comment by champred** [comment]
Just so I understand, your suggestion is that Pokemon with only one ability will have their hidden ability added as a secondary ability instead? If so I can possibly look into adding that next week.

**Comment by AdemAFN** [comment]
> Just so I understand, your suggestion is that Pokemon with only one ability will have their hidden ability added as a secondary ability instead? If so I can possibly look into adding that next week.

If it's the best possibly doable then yes.

**Comment by champred** [comment]
I've made a test release for the proposed feature you can try out [here](https://github.com/champred/UPR-Android/releases/tag/v0.5.2-rc1).
